header.go

```diff
@@ -513,6 +514,85 @@ func fixUnquotedSpecials(s string) string {
 	return clean.String()
 }
 
+func fixUnescapedQuotes(s string) string {
+	split := strings.SplitAfter(s, ";")
+	fresh := &strings.Builder{}
+	for i := 0; i < len(split); i++ {
+		// find the "="
+		idx := strings.IndexByte(split[i], '=')
+		if idx < 0 {
+			// no "=", must be the content-type or a comment
+			fresh.WriteString(split[i])
+			continue
+		}
+		fresh.WriteString(split[i][:idx])
+		param := split[i][idx:]
+		openingQuoteIdx := strings.IndexByte(param, '"')
+		closingQuoteIdx := strings.LastIndexByte(param, '"')
+		// cut out early if there are no quotes
+		if openingQuoteIdx < 0 && closingQuoteIdx < 0 {
+			// value isn't quoted
+			fresh.WriteString(param)
+			continue
+		}
+		// lets append the next chunk of split here in case of a semicolon mid string
+		if closingQuoteIdx == openingQuoteIdx {
+			param = fmt.Sprintf("%s%s", param, split[i+1])
+			closingQuoteIdx = strings.LastIndexByte(param, '"')
+			i++
+			if closingQuoteIdx == openingQuoteIdx {
+				return fresh.String()
+			}
+		}
+		// it's got quotes, lets put the k/v separator back in along with everything upto the first quote
+		fresh.WriteByte('=')
+		// quotes start here
+		fresh.WriteByte('"')
+		fresh.WriteString(param[1:openingQuoteIdx])
+		// get just the value, less the outer quotes
+		rest := param[closingQuoteIdx+1:]
+		// if there is stuff after the last quote then we should escape the first quote
+		if len(rest) > 0 && rest != ";" {
+			fresh.WriteString("\\\"")
+		}
+		param = param[openingQuoteIdx+1 : closingQuoteIdx]
+		alreadyEscaped := false
+		for strIdx := range param {
```

Comment:
I don't feel like the name `strIdx` conveys anything unique, and this loop is fairly short.

```suggestion
		for j := range param {
```